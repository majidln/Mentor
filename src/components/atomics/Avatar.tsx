import React, {FC, ReactElement} from 'react';
import {View, Image, StyleSheet} from 'react-native';

export interface AvatarProps {
  picture?: string;
}

const IMAGE_SIZE = 120;

export const AvatarComp: FC<AvatarProps> = ({picture = ''}): ReactElement => {
  if (!picture) {
    return <View />;
  }
  return (
    <View style={styles.imageWrapper}>
      <Image
        style={styles.image}
        source={{uri: picture}}
        resizeMode="contain"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  imageWrapper: {
    alignItems: 'center',
  },
  image: {
    width: IMAGE_SIZE,
    height: IMAGE_SIZE,
    borderRadius: IMAGE_SIZE / 2,
  },
});

function areEqual(prevProps: AvatarProps, nextProps: AvatarProps) {
  return prevProps.picture === nextProps.picture;
}
export const Avatar = React.memo(AvatarComp, areEqual);
