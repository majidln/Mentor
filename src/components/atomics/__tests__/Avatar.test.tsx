import 'react-native';
import React from 'react';
import {Image} from 'react-native';
import renderer from 'react-test-renderer';

import {Avatar} from './../Avatar';

describe('Test for atomic Avatar', () => {
  const picture = 'my_avatar';
  it('Avatar Image render correctly', () => {
    const testRenderer = renderer.create(<Avatar picture={picture} />);
    const testInstance = testRenderer.root;

    const element = testInstance.findByType(Image);

    expect(element.props.source).toEqual({uri: picture});
  });

  it('Avatar Snapshot renders correctly', () => {
    const tree = renderer.create(<Avatar picture={picture} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
