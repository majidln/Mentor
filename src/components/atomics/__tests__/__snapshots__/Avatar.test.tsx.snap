// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for atomic Avatar Avatar Snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "my_avatar",
      }
    }
    style={
      Object {
        "borderRadius": 60,
        "height": 120,
        "width": 120,
      }
    }
  />
</View>
`;
